pkg/controller/intrusiondetection/intrusiondetection_controller_test.go

```diff
@@ -418,6 +430,177 @@ var _ = Describe("IntrusionDetection controller tests", func() {
+			defaultTSEEInstanceKey := client.ObjectKey{Name: "tigera-secure"}
+
+			Expect(c.Create(ctx, &operatorv1.ManagementClusterConnection{
+				ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"},
+				Spec: operatorv1.ManagementClusterConnectionSpec{
+					ManagementClusterAddr: "127.0.0.1:12345",
+				},
+			})).ToNot(HaveOccurred())
+
+			_, err := r.Reconcile(ctx, reconcile.Request{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			result := &operatorv1.IntrusionDetection{}
+			err = c.Get(ctx, defaultTSEEInstanceKey, result)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			Expect(len(result.Spec.AnomalyDetection.StorageClassName)).To(Equal(0))
+		})
+
+		It("should not set AD Spec defaults if not originally not defined", func() {
+			defaultTSEEInstanceKey := client.ObjectKey{Name: "tigera-secure"}
+
+			_, err := r.Reconcile(ctx, reconcile.Request{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			result := &operatorv1.IntrusionDetection{}
+			err = c.Get(ctx, defaultTSEEInstanceKey, result)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			Expect(len(result.Spec.AnomalyDetection.StorageClassName)).To(Equal(0))
+		})
+
+		It("should degrade if AD Spec is set in managed cluster", func() {
+			Expect(c.Create(ctx, &operatorv1.ManagementClusterConnection{
+				ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"},
+				Spec: operatorv1.ManagementClusterConnectionSpec{
+					ManagementClusterAddr: "127.0.0.1:12345",
+				},
+			})).ToNot(HaveOccurred())
+
+			Expect(c.Delete(ctx, &operatorv1.IntrusionDetection{ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"}})).NotTo(HaveOccurred())
+
+			Expect(c.Create(ctx, &operatorv1.IntrusionDetection{
+				ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"},
+				Spec: operatorv1.IntrusionDetectionSpec{
+					AnomalyDetection: operatorv1.AnomalyDetectionSpec{
+						StorageClassName: expectedADStorageClassName,
+					},
+				},
+			})).ShouldNot(HaveOccurred())
+
+			_, err := r.Reconcile(ctx, reconcile.Request{})
+			Expect(err).Should(HaveOccurred())
+			Expect(mockStatus.AssertNumberOfCalls(nil, "SetDegraded", 1)).To(BeTrue())
+		})
+
+		It("should degrade if provided a non kubernets qualified AD storage name", func() {
+			Expect(c.Delete(ctx, &operatorv1.IntrusionDetection{ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"}})).NotTo(HaveOccurred())
+
+			malFormedStorageName := "malformed_storage_name"
+			Expect(c.Create(ctx, &operatorv1.IntrusionDetection{
+				ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"},
+				Spec: operatorv1.IntrusionDetectionSpec{
+					AnomalyDetection: operatorv1.AnomalyDetectionSpec{
+						StorageClassName: malFormedStorageName,
+					},
+				},
+			})).ShouldNot(HaveOccurred())
+
+			_, err := r.Reconcile(ctx, reconcile.Request{})
+			Expect(err).Should(HaveOccurred())
+			Expect(mockStatus.AssertNumberOfCalls(nil, "SetDegraded", 1)).To(BeTrue())
+		})
+
+		It("should degrade if ADSpec,StorageClassName is provided when type is Ephemeral", func() {
```

Comment:
This title isn't accurate anymore. Is there anything special this test is covering that isn't covered by the other tests?

Reply:
nope nothing new, just forgot to remove https://github.com/tigera/operator/pull/2094/commits/e393f5a8741e3e1206575f5c92123a92ebc837f9